src/lib/krad/packet.c

```diff
@@ -170,34 +180,24 @@ auth_generate_random(krb5_context ctx, uchar *rauth)
 /* Generate a response authenticator field. */
 static krb5_error_code
 auth_generate_response(krb5_context ctx, const char *secret,
-                       const krad_packet *response, const uchar *auth,
-                       uchar *rauth)
+                       const krb5_data *respbuf, const uint8_t *auth,
+                       uint8_t *rauth)
```

Comment:
I modified `auth_generate_response()` to accept a raw packet buffer because it is needed to find the matching request of a response, which is then used to verify the Message-Authenticator.